optuna/trial/_fixed.py

```diff
@@ -108,16 +108,23 @@ def suggest_discrete_uniform(self, name: str, low: float, high: float, q: float)
         discrete = distributions.DiscreteUniformDistribution(low=low, high=high, q=q)
         return self._suggest(name, discrete)
 
-    def suggest_int(self, name, low, high, step=1, log=False):
-        # type: (str, int, int, int, bool) -> int
-        if log:
-            sample = self._suggest(
-                name, distributions.IntLogUniformDistribution(low=low, high=high, step=step)
-            )
+    def suggest_int(self, name: str, low: int, high: int, step: int = 1, log: bool = False) -> int:
+        if step != 1:
+            if log:
+                raise ValueError("The parameter `step >= 2` is not supported when `log` is True.")
```

Comment:
How about aligning the error message to the actual logic? And this is a detail but how about also including the actual value in the error message as it usually helps debugging?

Lastly, just wondering but does the number 2 come from the base?

Reply:
Thanks for your suggestions. I will follow them.

The number 2 does not follow from the base of the logarithm. Here, the `step` is the discretization interval in the integer domain. So, `step` is greater than 2.